cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -189,15 +176,18 @@ impl YaccParser {
                     avoid_insert.insert(n);
                     i = self.parse_ws(j, true)?;
                 }
-                self.ast.avoid_insert = Some(avoid_insert);
+
+                if let Some(ref mut ai) = self.ast.avoid_insert {
```

Comment:
Might be simpler as something like:

```rust
if self.ast.avoid_insert.is_none() {
    self.ast.avoid_insert = Some(HashSet::new())
}
self.ast.avoid_insert.extend(avoid_insert);
```
?

Reply:
I did try that but was getting:
```
error[E0507]: cannot move out of borrowed content                                                                                                                                                                                                             
   --> cfgrammar/src/lib/yacc/parser.rs:181:21                                                                                                                                                                                                                
    |                                                                                                                                                                                                                                                         
181 |                     self.ast.avoid_insert.unwrap().extend(avoid_insert);                                                                                                                                                                                
    |                     ^^^^ cannot move out of borrowed content 
```

Reply:
That error suggests you need `self.ast.avoid_insert.as_ref().unwrap().extend(...)`.

Reply:
But I can replace the for loop. I actually didn't see HashSet had an extend method and tried to use `union` instead, which didn't work.